src/averageAges.js

```diff
@@ -59,6 +78,23 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+
+  const peopleWithMother = people.filter(person =>
+    ((people.some(child => child.name === person.mother)
+    && (onlyWithSon
+      ? person.sex === 'm'
+      : true)
+    ))
+  );
+
+  const mothersYears = peopleWithMother.map(person =>
+    ({
+      died: person.born,
```

Comment:
It's not a good idea, it's not `died` value
In this case we can write our own part of the code, without using `calculateAverageAge` function